docs/design/simplified_federation_deployment.md

```diff
@@ -0,0 +1,387 @@
+* Also request a dynamically provisioned persistent storage device for
+  the PVC.
+
+**6. Create federation API server**
+
+* Create a deployment for the federation API server in the `federation-
+  system` namespace.
+* The pod template spec is composed of two containers: one for `etcd`
+  and the other one for the federation API server.
+* The pod template spec references the PVC created in the previous step
+  as a volume and the `etcd` container mounts that volume to store
+  `etcd`'s data.
+* The federation API server container uses the official `hyperkube`
+  release image.
+* Since the API server is stateless, it needs no persistent storage.
+  However, the certificates and the credentials are made available to
+  the API server by mounting them as volumes inside the API server
+  container.
+
+**7. Create federation controller manager**
+
+* Create a deployment for the federation controller manager in the
+  `federation-system` namespace.
+* The pod template spec consists of a single container for the
+  federation controller manager and it also uses the same official
+  `hyperkube` release image as the federation API server.
+* Kubeconfig required by the federation controller manager to discover
+  and authenticate with the federation API server is referenced as a
+  secret in the pod template spec and mounted as a volume inside the
+  container.
+* The mounted kubeconfig file is passed as an argument to the federation
+  controller manager's `--kubeconfig` flag.
+
+##### Additional enhancements
+
+Users can pass `--register` as an additional flag to `kubeadm init
+federation` to also register the cluster with federation. See the
+[Cluster Registration/Deregistration](#cluster-
+registrationderegistration) section below for details on how this is
+done.
+
+##### Caveats
+
+* In the first phase of this implementation, we only create a single
+  instance of `etcd`. This neither makes `etcd` nor the federation
+  control plane HA. HA federation control plane is on the road map and
+  will arrive in one of the future releases.
+* Federation control plane components aren't started until additional
+  nodes join the bootstrap cluster. In other words, these components
+  aren't pinned to the bootstrap cluster's master node. We just create
+  their manifests at this stage and they are started when additional
+  schedulable nodes join the bootstrap cluster.
+
+
+#### Cluster Registration/Deregistration
+
+`kubectl` is Kubernetes' client-side CLI used for ongoing day-to-day
+interactions with a set of clusters. `kubectl` is already being extended
+to enable interactions with federation. Users can also already use
+`kubectl` to register clusters with and deregister clusters from
+federation. As already described, this process is manual, tedious and
+error-prone. Hence, we aim to reduce these processes to a single command
+each.
+
+##### Registration
+
+When a user runs `kubectl register <cluster-context-name> --bootstrap-
```

Comment:
The user who administrates a cluster isn't necessarily the one who joins a federation.  While I appreciate the simplification of setup, this seems to be the wrong order.

Why aren't these commands part of kubeadm?